### DISCUSSION TITLE
how do I change the highlighting of pragma regions

### DISCUSSION THREAD

**Discussion by Holo-Viktor** [microsoft/vscode-cpptools#7072] [Q&A] [answered]
```c
#pragma region xyz
void foo();
#pragma endregion xyz
```
Specifically I want to change the color of the `region` and the `xyz` parts.

**Comment by Colengms** [comment, marked as answer]
Hi @Holo-Viktor .  Colorization for content that can be identified syntactically is handled by VS Code's TextMate grammar for C/C++.  You might want to following up in [the repo for the TextMate grammar used by VS Code for C/C++](https://github.com/jeff-hykin/cpp-textmate-grammar).

**Reply by sean-mcmanus** [reply]
If you just want to change the region and the xyz part to a new color, you can use something like
```
    "editor.tokenColorCustomizations": {
        "textMateRules": [
            {
                "scope": [
                    "entity.other.attribute-name.pragma.preprocessor.c"
                ],
                "settings": {
                    "foreground": "#ff0000"
                }
            }
        ]
    }
```
in your settings.json, but the grammar isn't currently setup to handle the "region" part specially. The repo Colen mentioned would be responsible for adding that.

**Reply by Holo-Viktor** [reply]
@sean-mcmanus yes that worked, thanks. in my case i had to change the scope to `"entity.other.attribute-name.pragma.preprocessor.cpp"`

**Reply by sean-mcmanus** [reply]
Ah, good catch.
